### ISSUE TITLE
BigQuery pipe AGGREGATE/EXTEND clauses can't be indented (LT02 false positive)

### ISSUE THREAD

**Issue by ricardoalencar-pr** [sqlfluff/sqlfluff#8189]
# BigQuery pipe `AGGREGATE` / `EXTEND` clauses are not indentable (LT02 false positive)

## Dialect
`bigquery`

## Version
sqlfluff 4.2.2 (also reproduces on `main`)

## Description
In BigQuery [pipe query syntax](https://cloud.google.com/bigquery/docs/reference/standard-sql/pipe-syntax),
the `AGGREGATE` and `EXTEND` clauses take a comma-delimited list of
expressions, just like `SELECT` and `GROUP BY`. But unlike `SELECT` /
`GROUP BY`, their grammar emits no `Indent` / `Dedent` metadata around
the list. As a result, `LT02` (layout.indent) flags any continuation
line as *"Line should not be indented"*, and `sqlfluff fix` refuses to
indent them — so a multi-line `AGGREGATE` cannot be written in an
indented style at all.

## Expected Behaviour
A multi-line pipe `AGGREGATE` / `EXTEND` list indents like a `SELECT` or
`GROUP BY` list and lints clean.

## Observed Behaviour
```
$ sqlfluff lint --dialect bigquery example.sql
L:   3 | P:   1 | LT02 | Line should not be indented. [layout.indent]
L:   4 | P:   1 | LT02 | Line should not be indented. [layout.indent]
```

## Steps to Reproduce
Lint this with `--dialect bigquery`:
```sql
FROM my_table
|> AGGREGATE
    SUM(a) AS total_a,
    SUM(b) AS total_b
GROUP BY c
```

## Root cause
`AggregateClauseSegment` and `ExtendClauseSegment` in
`src/sqlfluff/dialects/dialect_bigquery.py` wrap their `Delimited(...)`
list without `Indent` / `Dedent`, whereas `GroupAndOrderByClauseSegment`
(the `GROUP BY` used within the same pipe syntax) does. Adding
`Indent` / `Dedent` around the two lists resolves it.
